feat: Add device onboarding type 'preconnect' (#17)
nRFCloud/device-simulator-v2: README.md

### Most basic usage
The most basic usage is just creating a device. For that you just need an API key. The device ID will be randomly generated and the rest of the necessary information (mqtt endpoint, cert, and mqtt prefix) will be pulled from the device API. 

> The host defaults to the production environment (https://api.nrfcloud.com). **Nordic Semiconductor personnel**: if you want to use the simulator on our `dev`, `feature`, or `beta` environments, set a `API_HOST` env var to `https://api.[dev|feature|beta].nrfcloud.com` (you will need an AWS account for that environment). You can also set the `API_HOST` env var to the url of your sub account (ie `https://api.<user_id>.nrfcloud.com`) and the `stage` will be automatically set to `dev` (the stage is used for the job subscription topics for FOTA). You no longer need to set a `STAGE` variable, and it will be ignored.

Reviewer: Why is this here? This is a public repo, and a while ago we scrubbed out any reference to other stages, including here I thought. Remove it, we can describe usages for other hosts elsewhere in private docs like Confluence

Author: Ahh, this was added from a merge conflict. Apologies.

> ### Associate device to your account

Reviewer: ```suggestion
### Onboard a device to your account
```

Reviewer: Also change some more "associate" --> "onboard" references earlier in this file.

> This will create a new device and associate it to the account for the API key.

Reviewer: ```suggestion
This example will onboard a device to nRF Cloud by creating a new simulated device and associating it to the user's team. The user's API key authenticates the REST request. A default device shadow will be created in the style of the Asset Tracker v2 (atv2) sample firmware. For the pre-connect mode of onboarding, the device simulator will create and provide self-signed certificates in the onboarding process.
```

```
npx @nrfcloud/device-simulator-v2 -k <api key> -a -t atv2 -a preconnect
```

### Simulate sensor outputs

> Include any combination of the options listed after `-s` below:

Reviewer: ```suggestion
After creating and associating your device, you can include any combination of the options listed after `-s` below, to simulate the device sending messages containing device sensor data:
```

```

> npx @nrfcloud/device-simulator-v2 -k <api key> -a preconnect -d <device ID you already onboarded> -s gps,acc,device,temp

Reviewer: ```suggestion
npx @nrfcloud/device-simulator-v2 -k <api key> -d <device ID you already onboarded> -s gps,acc,device,temp
```

```

> *Note! Usually the `-a` is not necessary, since the device is already onboarded to your account. However, due to bug IRIS-3450, this does not always work. Adding the `-a` is harmless and is a workaround for now.*

Reviewer 2: is this still true? doesn't this change the command to now become necessary  (telling the simulator to use the "jitp" or "preconnect" modality). It also seems IRIS-3450 is not reproducible anymore?

Reviewer: In the series of sections above, the device is already onboarded (associated) in the "Onboard device to your account" section. But maybe it's not obvious that this is a series of commands. Since IRIS-3450 is not an issue anymore (we think!), you can just remove the Note. But let's ensure people reading know it has to be associated/onboarded already, see above.
